src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
```

Comment:
> in most UTs the name is a full qualified one

UTs are specific just to our tests. Users mostly use the configuration file and our examples in XDOCs which just specify check name alone.
See:
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml
The names inside the `module` tag is what eventually gets called in this method to create the class instance.

> If we use your order without skipping, then we would do step 1, 2, 3 each time and it slows down test quite a lot.

1 and 2 should be pretty fast after the first run. They are only looking up names in the map and it should never cleared.